### ISSUE TITLE
Librechat permission issues and not picking up Ollama

### ISSUE THREAD

**Issue by FrantaNautilus** [av/harbor#35]
I am trying to use Librechat with `harbor` and there are two issues that prevent it from working. I am reporting these together, because I am not sure these are not linked. OS: Bluefin DX (Fedora Silverblue 40, with docker and docker compose)

#### sub-issue 1
When opening Librechat with `harbor librechat`, it intermediately crashes, here is the output of `harbor logs librechat`:
```
WARN[0000] The "HARBOR_WHISPER_HOST_PORT" variable is not set. Defaulting to a blank string. 
WARN[0000] The "HARBOR_WHISPER_VERSION" variable is not set. Defaulting to a blank string. 
harbor.librechat  | 
harbor.librechat  | > LibreChat@v0.7.5-rc2 backend
harbor.librechat  | > cross-env NODE_ENV=production node api/server/index.js
harbor.librechat  | 
harbor.librechat  | 2024-09-27 19:07:28 info: [Optional] Redis not initialized. Note: Redis support is experimental.
harbor.librechat  | 2024-09-27 19:07:29 error: There was an uncaught error: EACCES: permission denied, open '/app/api/logs/meiliSync-2024-09-27.log'
```
I tried to fix this by changing the ownership of the files in `.harbor/librechat` to my user (specifically `sudo chmod -R user:user .harbor/librechat`) and I was able to get Librechat started.

#### sub-issue 2
Librechat opened with `harbor up librechat ollama` does not pick up the running `ollama` and seams to have a problem finding the RAG container. Specifically, when the `librachat` is opened and Ollama provider is selected, there are no models listed. Additionally when a message is submitted expecting default model to be running, an error is displayed that `ollama` cannot be found.
Log from `harbor logs librechat`
```
WARN[0000] The "HARBOR_WHISPER_VERSION" variable is not set. Defaulting to a blank string. 
WARN[0000] The "HARBOR_WHISPER_HOST_PORT" variable is not set. Defaulting to a blank string. 
harbor.librechat  | 2024-09-27 19:11:46 warn: RAG API is either not running or not reachable at http://lcrag:33892, you may experience errors with file uploads.
harbor.librechat  | 2024-09-27 19:11:46 info: Updating 'USER' role MULTI_CONVO 'USE' permission from false to: true
harbor.librechat  | 2024-09-27 19:11:46 info: Updated 'USER' role permissions
harbor.librechat  | 2024-09-27 19:11:46 info: No changes needed for 'ADMIN' role permissions
harbor.librechat  | 2024-09-27 19:11:46 info: 
harbor.librechat  | Outdated Config version: 1.1.5
harbor.librechat  | Latest version: 1.1.7
harbor.librechat  | 
harbor.librechat  |       Check out the Config changelogs for the latest options and features added.
harbor.librechat  | 
harbor.librechat  |       https://www.librechat.ai/changelog
harbor.librechat  | 
harbor.librechat  | 
harbor.librechat  | 2024-09-27 19:11:46 info: Server listening on all interfaces at port 33891. Use http://localhost:33891 to access it
harbor.librechat  | {"level":"info","message":"[Login] [Login successful] [Username: user] [Request-IP: 172.19.0.1]","timestamp":"2024-09-27T19:13:15.650Z"}
harbor.librechat  | 2024-09-27 19:13:15 error: Failed to derive base URL Invalid URL
harbor.librechat  | 2024-09-27 19:13:15 error: Failed to fetch models from Ollama API. If you are not using Ollama directly, and instead, through some aggregator or reverse proxy that handles fetching via OpenAI spec, ensure the name of the endpoint doesn't start with `ollama` (case-insensitive). Cannot read properties of undefined (reading 'status')
harbor.librechat  | 2024-09-27 19:13:20 error: Failed to fetch models from SGLang API
harbor.librechat  | The request either timed out or was unsuccessful. Error message:
harbor.librechat  |  Cannot read properties of undefined (reading 'status')
harbor.librechat  | 2024-09-27 19:26:56 error: [handleAbortError] AI response error; aborting request: {"type":"no_user_key"}
harbor.librechat  | 2024-09-27 19:26:56 warn: Invalid conversation ID: null
```
Looking at the config file `librechat.yml`, it looks like the Librechat is expecting Ollama to be running at default URL, that being  `http://ollama:11434`, however in my case Ollama is automatically started on `http://localhost:33821`.

**Comment by av** [comment]
> When opening Librechat with `harbor librechat`, it intermediately crashes

This is happening because of the poorly arranged dependencies between sub-services, it wasn't caught during development as the compose file was created gradually, so the same problem wasn't surfaced

```bash
# Use this to fix all relevant volume permissions at once
harbor fixfs
# Clean previous data
rm -rf ./librechat/data
rm -rf ./librechat/images
rm -rf ./librechat/logs
rm -rf ./librechat/meili_data_v1.7
rm -rf ./librechat/vectordb
```
⚠️ In case you'll want to repeat - note that it removes all librechat data

With these, I was able to reproduce the issue you're describing and figured out a way to re-arrange the sub-services dependency order and UIDs. After doing that, I'm sure the [official compose setup](https://github.com/danny-avila/LibreChat/blob/main/deploy-compose.yml) is not free of these same issues either.

> Librechat opened with harbor up librechat ollama does not pick up the running ollama and seams to have a problem finding the RAG container.

Yes, apparently the environment variable interpolation stopped working, so none of the pre-configured connectivity was working as expected. I introduced a [config merging setup](https://github.com/av/harbor/wiki/6.-Harbor-Compose-Setup#config-merging) for librechat, now it should resolve the the env vars in the file correctly.

With these changes, I was able to:

```bash
harbor up librechat # ollama is the default
harbor open librechat

# Register default user
# Chat with ollama
```

![image](https://github.com/user-attachments/assets/85d111a2-99b3-4797-9a2a-3fb6f0a6757a)

I've published all these fixes as [v0.1.35](https://github.com/av/harbor/releases/tag/v0.1.35).

As usual - tremendous thanks for trying stuff out and for detailed reports!

**Comment by FrantaNautilus** [comment]
Thank you so much for detailed reply and quick fix. I confirm that Librechat now works with ollama without problems.
Just reading through the documentation and code feels like a masterclass in Docker and Bash.
